pahfit/model.py

```diff
@@ -0,0 +1,581 @@
+            param_info[3],
+            instrumentname,
+            update_fwhms=use_instrument_fwhm,
+            redshift=redshift,
+        )
+        # abs_info
+        param_info[4] = PAHFITBase.update_dictionary(
+            param_info[4], instrumentname, redshift
+        )
+
+        return param_info
+
+    def _backport_param_info(self, param_info):
+        """Convert param_info to values in features table.
+
+        Temporary hack to make the new system compatible with the old system.
+
+        TODO: if we remove the param_info stuff entirely, we won't need this
+
+        """
+        # unfortunately, there is no implementation for this, even in
+        # the original code. That one goes straight from astropy model
+        # to table... But we can do a kludge here: convert to model
+        # first, and then back to table.
+        astropy_model = PAHFITBase.model_from_param_info(param_info)
+        self._parse_astropy_result(astropy_model)
+
+    def _construct_astropy_model(
+        self, instrumentname, redshift, use_instrument_fwhm=True
+    ):
+        """Convert the features table into a fittable model.
+
+        Some nuances in the behavior
+        - If a line has a fwhm set, it will be ignored, and replaced by
+          the calculated fwhm provided by the instrument model.
+        - If a line has been masked by _parse_astropy_result, and this
+          function is called again, those masks will be ignored, as the
+          data range might have changed.
+
+        TODO: Make sure the features outside of the data range are
+        removed. The instrument-based feature check is done in
+        _kludge_param_info(), but the observational data might only
+        cover a part of the instrument range.
+
+        """
+        param_info = self._kludge_param_info(
+            instrumentname, redshift, use_instrument_fwhm
+        )
+        return PAHFITBase.model_from_param_info(param_info)
+
+    def _parse_astropy_result(self, astropy_model):
+        """Store the result of the astropy fit into the features table.
+
+        Every relevant value inside the astropy model, is written to the
+        right position in the features table.
+
+        For the unresolved lines, the widths are calculated by the
+        instrument model, or fitted when these lines are in a spectral
+        overlap region. The calculated or fitted result is written to
+        the fwhm field of the table. When a new model is constructed
+        from the features table, this fwhm value will be ignored.
+
+        For features that do not correspond to the data range, all
+        parameter values will be masked. Their numerical values remain
+        accessible by '.data' on the masked entity. This way, We still
+        keep their parameter values around (as opposed to removing the
+        rows entirely). When data with a larger range are passed for
+        another fitting call, those features can be unmasked if
```

Comment:
Right now there could be some confusion because "value masked" has a dual meaning. It can mean either "feature disabled" or "parameter not relevant for feature". (e.g. wavelength is masked out for the lines corresponding to a blackbody).

> Although it may be easy to just copy the features, delete the rows you don't want, create a Model from that, and fit again.

Actually, creating the model is not the problem here. The these masks are not used at all during model construction. In fact, the contents of the constructed model are used to *determine* which features need to be masked. They are only a signal to the user that the feature was not fit, and a nice visual cue when the table is printed.

But I agree that replacing this by an explicit "enabled" column, or only printing out those that were fitted, would make more sense to users.

Reply:
Good points; masking is used more to indicate "this one isn't relevant".  But if you start with a rich model like the explicit `pdr.yaml` with tons of lines, and want to turn some of the off, despite overlapping with your dataset, probably the easiest way is to remove them from the features table entirely.  I think that's fine.  You can always reread the `pdr.yaml` from disk.  For simplicity maybe we stick to that approach for now?